### PR TITLE
Use bigint for wei amounts

### PR DESCRIPTION
**Pull request by such**
Bump to 3.0.0

### CODE REVIEW

File: src/starkware/signature.js [hunk 1]
```diff
@@ -244,8 +244,8 @@ function hashLimitOrderMsgWithFee(
  ---------------
  @param {string|number} vaultSell - uint31 (as int)
  @param {string|number} vaultBuy - uint31 (as int)
- @param {string} amountSell - uint63 (as decimal string)
- @param {string} amountBuy - uint63 (as decimal string)
+ @param {bigint} amountSell - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amountSell - uint63
```

File: src/starkware/signature.js [hunk 2]
```diff
@@ -244,8 +244,8 @@ function hashLimitOrderMsgWithFee(
...
+ @param {bigint} amountBuy - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amountBuy - uint63
```

File: src/starkware/signature.js [hunk 3]
```diff
@@ -304,14 +302,14 @@ export function getLimitOrderMsgHash(
  ---------------
  @param {string|number} vaultSell - uint31 (as int)
  @param {string|number} vaultBuy - uint31 (as int)
- @param {string} amountSell - uint63 (as decimal string)
- @param {string} amountBuy - uint63 (as decimal string)
+ @param {bigint} amountSell - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amountSell - uint63
```

File: src/starkware/signature.js [hunk 4]
```diff
@@ -304,14 +302,14 @@ export function getLimitOrderMsgHash(
...
+ @param {bigint} amountBuy - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amountBuy - uint63
```

File: src/starkware/signature.js [hunk 5]
```diff
@@ -304,14 +302,14 @@ export function getLimitOrderMsgHash(
...
  @param {string} tokenSell - uint256 field element strictly less than the prime (as hex string with 0x)
  @param {string} tokenBuy - uint256 field element strictly less than the prime (as hex string with 0x)
  @param {number} nonce - uint31 (as int)
  @param {number} expirationTimestamp - uint22 (as int).
  @param {string|number} feeVaultId - uint31 (as int)
- @param {string} feeLimit - uint63 (as decimal string)
+ @param {bigint} feeLimit - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} feeLimit - uint63
```

File: src/starkware/signature.js [hunk 6]
```diff
@@ -380,7 +375,7 @@ export function getLimitOrderMsgHashWithFee(
  is defined by the application.
  Expected types:
  ---------------
- @param {string} amount - uint63 (as decimal string)
+ @param {bigint} amount - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amount - uint63
```

File: src/starkware/signature.js [hunk 7]
```diff
@@ -444,7 +438,7 @@ export function getTransferMsgHash(
 
  Expected types of fee info params:
  ---------------
- @param {string} amount - uint63 (as decimal string)
+ @param {bigint} amount - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} amount - uint63
```

File: src/starkware/signature.js [hunk 8]
```diff
@@ -454,7 +448,7 @@ export function getTransferMsgHash(
  @param {string|null|undefined} condition - uint256 field element strictly less than the prime (as hex string with 0x)
  @param {string} feeToken - uint256 field element strictly less than the prime (as hex string with 0x)
  @param {number|string} feeVaultId - uint31 (as int)
- @param {string} feeLimit - uint63 (as decimal string)
+ @param {bigint} feeLimit - uint63 (as decimal string)
```
Comment (lPernod-sorare) [inline src/starkware/signature.js]:
```suggestion
 @param {bigint} feeLimit - uint63
```

**Review by lPernod-sorare — APPROVED** [review]
Approved with minor nits on JSDoc comments

**Comment by lPernod-sorare** [comment]
> Bump to 3.0.0

If we're bumping to 3.0.0, I'd like to bundle a few dependency upgrades here as well:

- We can start using `@ethereumjs/wallet` (not released when we did 2.0.0)
- We can bump `@noble` and `@scure` packages for better deduplication
- We can upgrade micro-starknet to `@scure/starknet`:  https://github.com/paulmillr/scure-starknet/releases

**Comment by such** [comment]
@lPernod-sorare I already pushed on npm... Can you do the dependency upgrade as a patch? Should not be breaking right?

**Pull request merged**